AirCasting/Graph/GraphView.swift

```diff
@@ -36,35 +36,92 @@ struct GraphView<StatsViewModelType>: View where StatsViewModelType: StatisticsC
+                        ZStack(alignment: .topLeading) {
+                            if let selectedStream = selectedStream {
+                                Graph(stream: selectedStream,
+                                      thresholds: threshold,
+                                      isAutozoomEnabled: session.type == .mobile).onDateRangeChange { [weak graphStatsDataSource, weak statsContainerViewModel] range in
+                                        graphStatsDataSource?.dateRange = range
+                                        statsContainerViewModel?.adjustForNewData()
+                                      }
+                                StatisticsContainerView(statsContainerViewModel: statsContainerViewModel,
+                                                        threshold: threshold)
+                            }
 
-                }
-                NavigationLink(destination: HeatmapSettingsView(changedThresholdValues: threshold.rawThresholdsBinding)) {
-                    EditButtonView()
-                }
-                .padding(.horizontal)
-                .padding(.top)
-                
-                ThresholdsSliderView(threshold: threshold)
-                    .padding()
-                    // Fixes labels covered by tabbar
-                    .padding(.bottom)
+                        }
+                        HStack() {
+                            startTimeText
+                            Spacer()
+                            endTimeText
+                        }
+                        .foregroundColor(.aircastingGray)
+                        .padding(.horizontal, 5)
+                        NavigationLink(destination: HeatmapSettingsView(changedThresholdValues: threshold.rawThresholdsBinding)) {
+                            EditButtonView()
+                        }
+                        .padding(.horizontal)
+                        .padding(.top)
+                        
+                        ThresholdsSliderView(threshold: threshold)
+                            .padding()
+                            // Fixes labels covered by tabbar
+                            .padding(.bottom)
+                    }
             }
             Spacer()
         }
         .navigationBarTitleDisplayMode(.inline)
     }
+    
+    var startTimeText: some View {
+        startTime()
+    }
+    
+    var endTimeText: some View {
+        endTime()
+    }
+    
+    func procceding(session: SessionEntity) -> Bool {
+        var showGraph = false
+        guard session.allStreams != nil else { return false }
+        (session.allStreams!.forEach({ stream in
+            (stream.allMeasurements != []) ? (showGraph = true) : (showGraph = false)
+        }))
+        if showGraph {
+            return true
+        } else {
+            return false
+        }
+    }
+    
+    func startTime() -> Text {
+            let formatter = DateFormatter()
```

Comment:
creating a new `DateFormatter` is a CPU-heavy task, plase use other solution so you don't create a new formatter on each view reload